model/Core/Classes/Build.md

```diff
@@ -0,0 +1,74 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# Build
+
+## Summary
+
+Class that describes a build of software/artifacts.
+
+## Description
+
+A build is a representation of the process in which a piece of software or
+artifact is built. It encapsulates information related to a build process and
+provides an element from which relationships can be created to relate the build
+to the inputs/outputs, and entities related to the build (e.g. builders,
+identities, etc.).
+
+Definitions of "BuildType", "ConfigSource", "Parameters" and "Environment" follow
+those defined in [SLSA provenance](https://slsa.dev/provenance/v0.2).
+
+<!--
+The build will include relationships such as:
+
+<Build> BUILT_BY <SPDX_Element> 
+<Build> BUILD_INITIATED_BY <Subject/Actor>
+<Package> BUILD_INPUT_OF <Build>
+<Package> BUILD_OUTPUT_OF <Build>
+
+For BUILT_BY:
+At the moment it can point to an artifact to signify a builder, but can point
+to a "Provider/Service" element in the future which can encapsulate more
+details of the builder if needed.
+-->
+
+## Metadata
+
+- name: Build
+- SubclassOf: Element
+- Instantiability: Concrete
+
+## Properties
+
+- buildType
+  - type: xsd:string
+  - minCount: 0
+- configSoureEntrypoint
+  - type: xsd:string
+  - minCount: 0
+- configSoureUri
+  - type: xsd:string
+  - minCount: 0
+- configSoureDigest
+  - type: xsd:string
+  - minCount: 0
+- parameters
+  - type: xsd:map<string>string
+  - minCount: 0
+- buildInvocationId
+  - type: xsd:string
+  - minCount:0
+  - maxCount:1
+- startTime
+  - type: xsd:DateTime
+  - minCount:0
+  - maxCount:1
+- endTime
+  - type: xsd:DateTime
+  - minCount:0
+  - maxCount:1
+- logReference
```

Comment:
`Build` already has an `externalReferences` property that you could put a reference in, however, for logs I would probably lean towards this being a reference to a file, that allows you to capture the hashes so you can verify the log hasn't been tampered with.

Reply:
My gut reaction is to have a relationship (eg. `LOG_OF`) for this, since there might be multiple log files or multiple log formats. Again, you can communicate hashes in the relationship's target Element.

Reply:
My gut reaction is to have a relationship (eg. LOG_OF) for this, since there might be multiple log files or multiple log formats. Again, you can communicate hashes in the relationship's target Element.

Reply:
quick question: if we want to point a log URI, what would be the best way to represent that? Like if the build produces a pointer to a URI that the logs will be accessible at (but it may not be immediately accessible or only accessible to authorized individuals)